liquibase-standard/src/main/java/liquibase/sqlgenerator/core/InsertGenerator.java

```diff
@@ -67,8 +68,10 @@ public void generateValues(StringBuilder sql,InsertStatement statement, Database
 
         for (String column : statement.getColumnValues().keySet()) {
             Object newValue = statement.getColumnValues().get(column);
-            if ((newValue == null) || "NULL".equalsIgnoreCase(newValue.toString())) {
+            if (newValue == null) {
```

Comment:
Hi @catull,

I've been testing this, and I see that it does not seem to work in all cases. For example, for enumeration types, this wouldn't be valid because it will try to insert a value other than the ones valid for that given enumeration. Not sure, if there is any other type this could also affect, that I haven't realized yet.

Reply:
Hmm, do you have a test case ?  I am not sure I can follow you.

Reply:
Yeap, I faced this issue in the `included.changelog.xml` (which is under this path `changelogs/pgsql/complete/included.changelog.xml`). The execution of the below changeset:

```xml
 <changeSet id="6" author="molivasdat">
      <loadData file="datafiles/enum-data.csv" relativeToChangelogFile="true" tableName="sample_enum_table">
          <column header="int_col" name="int_col" type="NUMERIC"/>
          <column header="str_col" name="str_col" type="STRING"/>
          <column header="enum_col" name="enum_col" type="OTHER"/>
      </loadData>
</changeSet>
```
This was making some integration tests fail because of trying to insert a `"NULL"` into an enum type column.

Reply:
@MalloD12

Can you be a bit more specific ?
There are about ~20 references to `included.changelog.xml`.

I'd like to investigate the test failure you refer to.

Thanks.

Reply:
At any rate, I found the "bug".

`LoadDataChange#generateStatementsFromRows()` conflates `null` and `"NULL"`.
Both would be used as a key in a `Map`.

In order to disinguish `null` from the String `"NULL"`, I introduce a `Null` class.

Please review this, @MalloD12.

Reply:
Hey @catull,

Are you able to see the this [execution details](https://github.com/liquibase/liquibase/actions/runs/12287878906/job/34393355812?pr=6506), you can find here what's currently failing.  If I'm not wrong, yes previously, were `PostgreSQLIntegrationTest` failing.

Reply:
```
Error:    LoadDataOtherTest.testOtherColumnLoadData:46 expected:<NULL> but was:<null>
```

I do not get this error locally 
![image](https://github.com/user-attachments/assets/c5bfd83e-4ccb-4058-9126-7cbdef8bd3c5)

Same deal with

```
Error:    LoadDataChangeTest.NULL placeholder handling:809 Condition not satisfied:
```

Runs smoothly locally:

![image](https://github.com/user-attachments/assets/a3ea66bf-ec8d-4daa-b130-426ab1bb901a)

Reply:
In fact, the whole test class `LoadDataChangeTest` runs all tests successfully.

Reply:
![image](https://github.com/user-attachments/assets/b03647bc-71e5-4419-ac1b-aa613fe1b969)

Reply:
The problem with OpenJDK 8 is, I cannot even build it on macOS, never have been.

I perform a `mvn clean package`, and I always end up with this:

![image](https://github.com/user-attachments/assets/620a3c79-e5f4-403c-9322-3747690f74c4)

Even after I `rm -fr ~/.m2/repository/jakarta`, the build never succeeds, same result as just shown.
I have stopped using OpenJDK 8 with Liquibase, when I develop with it.
I can succesfully build Liquibase with OpenJDK 11 and higher.

I use the official distribution of Liquibase in production with OpenJDK 8, OpenJDK 17 and OpenJDK 21.

Reply:
Even this does not help at all: `mvn -DskipTests -Dmaven.javadoc.skip=true clean package`, same error.

Reply:
This statement is from `changelog.txt`:

```
- (#5810) Bump jakarta.enterprise:jakarta.enterprise.cdi-api from 4.0.1 to 4.1.0 @dependabot
```

That version is from March 2024, when the project `jakarta.enterprise.cdi-api` jumped from JDK 11 -> JDK 17.

So, which trick do you use to build Liquibase under JDK 8 ?

Reply:
Thanks, @MalloD12 

I had a look at a full test run under Fedora.
I believe I corrected them all now.

I get no errors now with `mvn -DtrimStackTrace=false -pl '!liquibase-cdi-jakarta' test`.